packages/io/3mf-deserializer/src/model.js

```diff
@@ -0,0 +1,66 @@
+import { maths } from '@jscad/modeling'
+
+import { parse } from './parse.js'
+
+/*
+ * Parse the given 3MF source (XML) into 3MF build items, objects and materials.
+ *
+ * Each build item is a 3MF mesh with
+ *   oid: object id of which to build (and transform)
+ *   transform : mat4 with transforms for rotation/translation
+ *   sequence: unique sequence number of item
+ */
+export const parseModel = (options, source) => {
+  const { includedItems } = options
+
+  // parse the 3MF contents (XML)
+  const { objects, materials, colorgroups, items } = parse(source)
+  if (objects.length === 0) {
+     throw new Error('3MF parsing failed, no valid 3MF objects retrieved')
+   }
+
+  // get a list of objects to include in the results
+  const buildItems = []
+  items.forEach((item) => {
+    const components = getComponents(item, objects)
+    for (let i = 0; i < components.length; i++) {
+      const component = components[i]
+      component.sequence = buildItems.length
+      buildItems.push(component)
+    }
+  })
+
+  if (includedItems === 'all') {
+    // FIXME loop through objects.. include anything missing
+  }
+
+  // console.log(model)
```

Comment:
remove comments?

Reply:
@platypii thanks for the review.

Documentation has been improved, comments in code removed, and JSDOC configured.